rename insomnia-app/app to insomnia/src

dimitropoulos opened the pull request:
As the title states, this PR has a very very tiny scope (despite the number of lines touched).  The first commit renames the directory and absolutely literally nothing else.

The second commit just handles all the places where the location was hardcoded.

The motivation for this PR is to bring some long-overdue consistency to simplify the building of this one package in relation to all other packages.  We have a quite consistent build process (with respect to project structure) _EXCEPT_ for this.

Then, due to [recent changes](https://github.com/Kong/insomnia/pull/4704) with electron builder, the 3rd and 4th commit handle renaming the `insomnia-app` directory to `insomnia`.

DMarby commented on packages/insomnia-smoke-test/playwright/paths.ts:
```diff
@@ -22,8 +22,8 @@ const electronBinary = path.join('node_modules', '.bin', process.platform === 'w
 
-// NOTE: main.min.js is built by app-build:smoke in /build and also by the watcher in /app
-export const mainPath = path.join(process.env.BUNDLE === 'dev' ? 'app' : 'build', 'main.min.js');
+// NOTE: main.min.js is built by app-build:smoke in insomnia-app/build and also by the watcher in insomnia-app/app
```
```suggestion
// NOTE: main.min.js is built by app-build:smoke in insomnia-app/build and also by the watcher in insomnia-app/src
```

dimitropoulos committed:
git mv ./packages/insomnia-app/app ./packages/insomnia-app/src/

only other change is two .gitignore files that were updated (for obvious reasons)

dimitropoulos committed:
manually renames all places app was hardcoded

dimitropoulos committed:
git mv ./packages/insomnia-app/ ./packages/insomnia

only other change is two .gitignore files that were updated (for obvious reasons)

dimitropoulos committed:
manual renames for insomnia-app and "insomnia app"

dimitropoulos committed:
fixes

ideally, this would be filtered in with the prior commits, but it was a real bear.

marckong reviewed (approved):
I ran some smoke tests on this branch, and build. I could run successfully the following scripts:
- app-run
- app-package
- app-build
- type-check

I wonder how this would impact [this effort](https://github.com/Kong/insomnia/pull/4713), so do you think we may wait until that PR gets merged?

DMarby commented on lerna.json:
```diff
@@ -3,7 +3,8 @@
     "packages/*",
-    "plugins/*"
+    "plugins/*",
+    "packages/insomnia/build"
```
This no longer exists since we merged the packedDependencies removal PR
```suggestion
```

dimitropoulos replied:
ahhhhhhh I thought it was the other way around.  good catch: e0d1ebd665e6cadc940ef5f0e406f1414907dcfd

DMarby commented on packages/insomnia-inso/src/jest/setup.ts:
```diff
@@ -1 +1 @@
-process.env.DEFAULT_APP_NAME = process.env.DEFAULT_APP_NAME || 'insomnia-app';
+process.env.DEFAULT_APP_NAME = process.env.DEFAULT_APP_NAME || 'insomnia';
```
This should not be changed, it refers to the user data directory on disk, not the package name.

```suggestion
process.env.DEFAULT_APP_NAME = process.env.DEFAULT_APP_NAME || 'insomnia-app';
```

dimitropoulos replied:
!!!!!!!!!!!!!! HOLY CRAP GOOD FIND 3eead72366259b84a03082623378d956db6a1560

DMarby commented on packages/insomnia-inso/webpack/webpack.config.development.js:
```diff
@@ -9,7 +9,7 @@ module.exports = merge(base, {
     new webpack.DefinePlugin({
       __DEV__: true,
-      'process.env.DEFAULT_APP_NAME': JSON.stringify('insomnia-app'),
+      'process.env.DEFAULT_APP_NAME': JSON.stringify('insomnia'),
```
This should not be changed, it refers to the user data directory on disk, not the package name.
```suggestion
      'process.env.DEFAULT_APP_NAME': JSON.stringify('insomnia-app'),
```

dimitropoulos replied:
3eead72366259b84a03082623378d956db6a1560

DMarby commented on packages/insomnia/src/main.development.ts:
```diff
@@ -41,7 +41,7 @@ if (envDataPath) {
   // Explicitly set userData folder from config because it's sketchy to rely on electron-builder to use productName, which could be changed by accident.
   const defaultPath = app.getPath('userData');
-  const newPath = path.join(defaultPath, '../', isDevelopment() ? 'insomnia-app' : appConfig.userDataFolder);
+  const newPath = path.join(defaultPath, '../', isDevelopment() ? 'insomnia' : appConfig.userDataFolder);
```
```suggestion
  const newPath = path.join(defaultPath, '../', isDevelopment() ? 'insomnia-app' : appConfig.userDataFolder);
```

dimitropoulos replied:
3eead72366259b84a03082623378d956db6a1560

dimitropoulos committed:
fixes lerna-app

dimitropoulos committed:
fixes insomnia-app data directory
